src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs

```diff
@@ -500,5 +504,37 @@ public bool ShouldGetByName(string resourceGroupName, string name)
 
             return false;
         }
+
+        private event EventHandler<StreamEventArgs> _writeDebugEvent;
+        private event EventHandler<StreamEventArgs> _writeVerboseEvent;
+        private event EventHandler<StreamEventArgs> _writeWarningEvent;
+
+        private void InitializeEventHandlers()
+        {
+            _writeDebugEvent -= WriteDebugSender;
+            _writeDebugEvent += WriteDebugSender;
+            _writeVerboseEvent -= WriteVerboseSender;
+            _writeVerboseEvent += WriteVerboseSender;
+            _writeWarningEvent -= WriteWarningSender;
+            _writeWarningEvent += WriteWarningSender;
+            AzureSession.Instance.RegisterComponent(WriteDebugKey, () => _writeDebugEvent);
```

Comment:
So, these aren't guaranteed to return on the same thread, as we discussed.  In the case of Debug traces, I thinkw e can write any messages to the existing queue, where they are drained at the dn of processing.  In the case of The Warning we write for the device authentication flow, I think we will need to handle that separately, in the code that is waiting for that authentication call.